v2/spanner-to-sourcedb/src/main/java/com/google/cloud/teleport/v2/templates/dbutils/connection/CassandraConnectionHelper.java

```diff
@@ -0,0 +1,202 @@
+   *     provided.
+   */
+  @Override
+  public synchronized void init(ConnectionHelperRequest connectionHelperRequest) {
+    if (connectionPoolMap != null && !connectionPoolMap.isEmpty()) {
+      LOG.info("Connection pool is already initialized.");
+      return;
+    }
+
+    LOG.info(
+        "Initializing Cassandra connection pool with size: {}",
+        connectionHelperRequest.getMaxConnections());
+
+    List<Shard> shards = connectionHelperRequest.getShards();
+    for (Shard shard : shards) {
+      if (!(shard instanceof CassandraShard)) {
+        LOG.error("Invalid shard type: {}", shard.getClass().getSimpleName());
+        throw new IllegalArgumentException("Invalid shard object");
+      }
+
+      CassandraShard cassandraShard = (CassandraShard) shard;
+      try {
+        CqlSession session = createCqlSession(cassandraShard);
+        String connectionKey = generateConnectionKey(cassandraShard);
+        connectionPoolMap.put(connectionKey, session);
+        LOG.info("Connection initialized for key: {}", connectionKey);
+      } catch (Exception e) {
+        LOG.error("Failed to initialize connection for shard: {}", cassandraShard, e);
+      }
+    }
+  }
+
+  /**
+   * Retrieves a {@link CqlSession} connection from the connection pool.
+   *
+   * @param connectionRequestKey The unique key identifying the connection in the pool.
+   * @return The {@link CqlSession} instance associated with the given key.
+   * @throws ConnectionException If the connection pool is not initialized or no connection is found
+   *     for the key.
+   */
+  @Override
+  public CqlSession getConnection(String connectionRequestKey) throws ConnectionException {
+    if (connectionPoolMap == null || connectionPoolMap.isEmpty()) {
+      LOG.warn("Connection pool not initialized.");
+      throw new ConnectionException("Connection pool is not initialized.");
+    }
+
+    CqlSession session = connectionPoolMap.get(connectionRequestKey);
+    if (session == null) {
+      LOG.error("No connection found for key: {}", connectionRequestKey);
+      throw new ConnectionException(
+          "No connection available for the given key: " + connectionRequestKey);
+    }
+
+    return session;
+  }
+
+  /**
+   * Checks if the connection pool is initialized and contains connections.
+   *
+   * @return {@code true} if the connection pool is initialized and not empty; {@code false}
+   *     otherwise.
+   */
+  @Override
+  public boolean isConnectionPoolInitialized() {
+    return connectionPoolMap != null && !connectionPoolMap.isEmpty();
+  }
+
+  /**
+   * Creates a {@link CqlSession} for the given {@link CassandraShard}.
+   *
+   * @param cassandraShard The shard containing connection details.
+   * @return A {@link CqlSession} instance.
+   */
+  private CqlSession createCqlSession(CassandraShard cassandraShard) {
+    CqlSessionBuilder builder = CqlSession.builder();
```

Comment:
Why can't we use [withConfigLoader](https://docs.datastax.com/en/latest-java-driver-api/com/datastax/oss/driver/api/core/session/SessionBuilder.html#withConfigLoader-com.datastax.oss.driver.api.core.config.DriverConfigLoader-)?

Reply:
Address the same @VardhanThigle removing duplicate line